packages/protocol/congestioncontrol/icca/scheduler/scheduler.go

```diff
@@ -412,13 +434,12 @@ loop:
 		select {
 		// every rate time units
 		case <-s.ticker.C:
-			// TODO: pause the ticker, if there are no ready blocks
 			if block := s.schedule(); block != nil {
+				//s.ticker.Reset(s.optsRate * time.Duration(block.Work()))
```

Comment:
Comment was part of another TODO which was left in accidentally. Essentially we would like to make the schedule work in units of work, not in number of blocks, so this requires a different wait time after each schedule, no a fixed wait time.
As for the removed TODO, you can't pause a ticker, you can only stop it and then it can't be restarted, so some different approach is required for that as well.
I will add a new TODO to capture all changes we would like to make.

Reply:
```suggestion
// TODO: make this operate in units of work, with a variable pause between scheduling depending on work scheduled.
	// TODO: don't use a ticker. Switch to a simple timer instead, and use a flag when ready to schedule something if there is nothing ready to be scheduled yet.
	// TODO: implement a token bucket for the scheduler to account for bursty arrivals.
```